vignettes/tcplfit2-vignette.Rmd

```diff
@@ -362,3 +362,148 @@ The output table resulting from the previous code chunk is the same format as th
+```
+
+```{r example 5 new lower-res, echo=FALSE}
+# print out the new results
+res_low2['Min. Conc.'] <- min(conc2)
+res_low2['Name'] <- name
+res_low2[1, c("Min. Conc.", "bmd", "bmdl", "bmdu")] <- round(res_low2[1, c("Min. Conc.", "bmd", "bmdl", "bmdu")], 3)
+DT::datatable(res_low2[1, c("Name",'Min. Conc.',"bmd", "bmdl", "bmdu")],rownames = FALSE)
+```
+
+```{r example 5 lower cont., echo=FALSE}
+logc_plot <- seq(from=-3,to=2,by=0.05)
+conc_plot <- 10**logc_plot
+
+plot(c(1,1),type="n",xlab="conc (uM)",ylab="Response",xlim=c(0.001,100),ylim=c(0,100),
+       log="x",main=paste(name,"\n",assay),cex.main=0.9)
+
+lines(c(res_low$bmd,res_low$bmd),c(0,50),col="green",lwd=2)
+rect(xleft=res_low$bmdl,ybottom=0,xright=res_low$bmdu,ytop=50,col=rgb(0,1,0, alpha = .5), border = NA)
+
+lines(c(res_low2$bmd,res_low2$bmd),c(0,50),col="blue",lwd=2)
+rect(xleft=res_low2$bmdl,ybottom=0,xright=res_low2$bmdu,ytop=50,col=rgb(0,0,1, alpha = .5), border = NA)
+
+points(res_low$bmd, 0, pch = "x", col = "green")
+points(res_low2$bmd, 0, pch = "x", col = "blue")
+points(conc2, resp2)
+lines(conc_plot, exp4(ps = c(res_low$tp, res_low$ga), conc_plot))
+```
+
+**Figure x**: This plot shows how BMD and its boundaries changed after "censoring". The green line and green "X" mark the BMD before "censoring" and the green column shows the original boundaries. The blue line and blue "X" mark the "censored" BMD and the blue column shows the updated boundaries. The BMD and its boundaries were forced to be shifted right because it's not allowed to be lowered than 0.48. Note that the updated boundaries also became narrower.  
+
+Let's consider another situation when BMD is higher than the maximum concentration in the data. This is not ideal because the experiments are typically designed as that anything beyond the maximum concentrate would cause server damage. If this happen, `bmd_up_bnd` can be used to restrict the BMD to be no higher than a percentage of the maximum dose. 
+```{r example 5 upper}
+# load example data
+spid <- unique(mc3$spid)[26]
+temp <- mc3[is.element(mc3$spid,spid),]
+
+# The data file has stored concentration in log10 form, so fix that
+conc <- 10**temp$logc # back-transforming concentrations on log10 scale
+resp <- temp$resp
+
+# pull out all of the chemical identifiers and the name of the assay
+dtxsid <- temp[1,"dtxsid"]
+casrn <- temp[1,"casrn"]
+name <- temp[1,"name"]
+assay <- temp[1,"assay"]
+
+# create the row object
+row <- list(conc = conc, resp = resp, bmed = 0, cutoff = cutoff, onesd = onesd,assay=assay,
+            dtxsid=dtxsid,casrn=casrn,name=name)
+
+# run the concentration-response modeling for a single sample
+res_up <- concRespCore(row,fitmodels = c("cnst", "hill", "gnls", "poly1", "poly2", 
+                                         "pow", "exp2", "exp3", "exp4", "exp5"), 
+                       conthits = T, aicc = F, bidirectional=F)
+
+concRespPlot(res_up,ymin=-10,ymax=100)
```

Comment:
May consider adjusting the y-axis for plots to be most appropriate to the example at hand.